constraint/pkg/client/drivers/k8scel/driver.go

```diff
@@ -48,69 +51,77 @@ import (
 //   Other friction points are commented with the keyword FRICTION.
 
 const (
-	Name = "K8sNativeValidation"
+	variablesKey = ctxKey("variables")
 
 	runTimeNS            = "runTimeNS"
 	runTimeNSDescription = "the number of nanoseconds it took to evaluate the constraint"
 )
 
+type ctxKey string
+
 var _ drivers.Driver = &Driver{}
 
 type Driver struct {
-	mux            sync.RWMutex
-	validators     map[string]validatingadmissionpolicy.Validator
-	filterCompiler cel.FilterCompiler
-	gatherStats    bool
+	mux         sync.RWMutex
+	validators  map[string]validatingadmissionpolicy.Validator
+	gatherStats bool
 }
 
 func (d *Driver) Name() string {
-	return Name
+	return pSchema.Name
 }
 
 func (d *Driver) AddTemplate(ctx context.Context, ct *templates.ConstraintTemplate) error {
 	if len(ct.Spec.Targets) != 1 {
 		return errors.New("wrong number of targets defined, only 1 target allowed")
 	}
-	var rawCode map[string]interface{}
+
+	var source *pSchema.Source
 	for _, code := range ct.Spec.Targets[0].Code {
-		if code.Engine != Name {
+		if code.Engine != pSchema.Name {
 			continue
 		}
-		objMap, ok := code.Source.Value.(map[string]interface{})
-		if !ok {
-			return errors.New("K8sNativeValidation code malformed")
+		var err error
+		source, err = pSchema.GetSource(code)
+		if err != nil {
+			return err
 		}
-		rawCode = objMap
 		break
 	}
-	if rawCode == nil {
+	if source == nil {
 		return errors.New("K8sNativeValidation code not defined")
 	}
 
-	validatorCode := &admissionv1alpha1.ValidatingAdmissionPolicy{}
-	if err := runtime.DefaultUnstructuredConverter.FromUnstructuredWithValidation(rawCode, validatorCode, true); err != nil {
+	// FRICTION: Note that compilation errors are possible, but we cannot introspect to see whether any
```

Comment:
:(

Reply:
Yeah, K8s will probably need to fix that themselves once they have status reporting.

Reply:
When they add status reporting, I mean